Bottom: 3c225ba09afc7892212755fe59eaa3c81def128d
Top:    3b4fc5224bc1896fb2302da84fb31b3448a0389d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-15 19:22:07 +0100

Refresh of try-something-with-the-rtrees

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 30d6e12..c7bf0b3 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -38,7 +38,7 @@
       all cases where original (Klamer Schutte) code is present
       are marked
 */
-#define NDEBUG
+//#define NDEBUG
 #include	<assert.h>
 #include	<stdlib.h>
 #include	<stdio.h>
@@ -101,7 +101,7 @@ if (((ptr) = malloc(sizeof(type))) == NULL) \
 #undef DEBUG_LABEL
 #undef DEBUG_ALL_LABELS
 #undef DEBUG_JUMP
-#undef DEBUG_GATHER
+#define DEBUG_GATHER
 #undef DEBUG_ANGLE
 #undef DEBUG
 #ifdef DEBUG
@@ -809,7 +809,7 @@ static int
 intersect (jmp_buf * jb, POLYAREA * b, POLYAREA * a, int add)
 {
   POLYAREA *t;
-  PLINE *pa, *pb;
+  PLINE *pa;
   contour_info c_info;
 
   /* Search the r-tree of the object with most contours
@@ -1709,7 +1709,7 @@ M_POLYAREA_Collect (jmp_buf * e, POLYAREA * afst, POLYAREA ** contours,
 	parent = NULL;
       for (cur = &a->contours; *cur != NULL; cur = next)
 	{
-          POLYAREA *tmp;
+          PLINE *tmp;
           tmp = *cur;
 	  next = &((*cur)->next);
 #warning PERHAPS WE NEED TO REMOVE THE CONTOUR FROM THE RTREE BECORE CALLING THIS?
@@ -1720,7 +1720,10 @@ M_POLYAREA_Collect (jmp_buf * e, POLYAREA * afst, POLYAREA ** contours,
             {
               next = cur;
 #warning SHOULD BE IN cntr_Collect??
+              if (tmp != parent)
               r_delete_entry ((rtree_t *)a->contour_tree, (BoxType *)tmp);
+//              if (tmp = parent)
+//                r_insert_entry ((rtree_t *)a->contour_tree, (BoxType )tmp);
             }
 	}
     }
